lib/src/interface/http_server.dart

```diff
@@ -0,0 +1,91 @@
+import 'dart:async';
+import 'dart:io';
+
+import 'package:seltzer/src/http_server_impl.dart';
+import 'package:seltzer/src/interface/http_request.dart';
+import 'package:seltzer/src/interface/socket.dart';
+
+/// An HTTP server that emits HTTP requests and socket connections.
+abstract class SeltzerHttpServer {
```

Comment:
Maybe just make this `implements Stream<ServerSeltzerHttpRequest>`?

You can remove the `requests` field and just do:

```dart
server.listen((request) { ... });
```